test/unit-test/CMakeLists.txt

```diff
@@ -1,89 +1,117 @@
+set( CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin )
+set( CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib )
+set( CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib )
+
+# ====================== coreHTTP library configurations =======================
+
+# Include filepaths for source and include.
+include( ${MODULE_ROOT_DIR}/httpFilePaths.cmake )
+
+# Define a http_parser resource path.
+set( HTTP_PARSER_DIR ${MODULE_ROOT_DIR}/source/third-party/http_parser CACHE INTERNAL "http-parser library source directory." )
+
+# Add the http_parser submodule to HTTP client library. It is out of the
+# unit-test tree and therefor needs it's own binary directory.
+add_subdirectory( ${MODULE_ROOT_DIR}/source/third-party ./build-http-parser )
+
+# Check if the http_parser source directory exists.
+if( NOT EXISTS ${HTTP_PARSER_DIR}/http_parser.c )
+    # Attempt to clone http_parser.
+    if( ${BUILD_CLONE_SUBMODULES} )
+        clone_http_parser()
+    else()
+        message( FATAL_ERROR "The required submodule http_parser does not exist. Either clone it manually, or set BUILD_CLONE_SUBMODULES to 1 to automatically clone it during build." )
+    endif()
+endif()
+
+# HTTP library target.
+add_library( http
+             ${HTTP_SOURCES} )
+
+# HTTP public include path.
+target_include_directories( http PUBLIC ${HTTP_INCLUDE_PUBLIC_DIRS} )
+
+# HTTP private include path. The unit test is default build so we include the test
+# configuration for now.
+target_include_directories( http PRIVATE ${HTTP_INCLUDE_PRIVATE_DIRS}
+                            ${MODULE_ROOT_DIR}/test/unit-test )
+
+# Link http_parser to http target.
+target_link_libraries( http PRIVATE http_parser )
+
+# Organization of HTTP in IDE projects.
+set_target_properties( http PROPERTIES FOLDER libraries/standard )
+source_group( include FILES ${MODULE_ROOT_DIR}/source/include/core_http_client.h source/portable/transport_interface.h )
+source_group( source FILES ${HTTP_SOURCES} )
+source_group( source\\private FILES ${MODULE_ROOT_DIR}/source/private/http_internal.h )
+
+# ============================ Test configurations =============================
+
+# Define a CMock resource path.
+set( CMOCK_DIR ${MODULE_ROOT_DIR}/test/unit-test/CMock CACHE INTERNAL "CMock library source directory." )
+
+# Include CMock build configuration.
+include( ${MODULE_ROOT_DIR}/test/unit-test/cmock_build.cmake )
+
+# Check if the CMock source directory exists, and if not present, clone the submodule
+# if BUILD_CLONE_SUBMODULES configuration is enabled. 
+if( NOT EXISTS ${CMOCK_DIR}/src )
+    # Attempt to clone CMock.
+    if( ${BUILD_CLONE_SUBMODULES} )
+        clone_cmock()
+    else()
+        message( FATAL_ERROR "The required submodule CMock does not exist. Either clone it manually, or set BUILD_CLONE_SUBMODULES to 1 to automatically clone it during build." )
+    endif()
+endif()
+
+# Add unit test and coverage configuration only if CMock directory exists.
+if( EXISTS ${CMOCK_DIR}/src )
```

Comment:
I realized that this if-check is not required as the Cmake build will halt with the `FATAL_ERROR` when the submodule clone is not found.
Thus, the if-check can be removed

Reply:
Oh right we can just rely on that error.